IQSS/dataverse-frontend
# Feat/425 publish dataset and collection

## What this PR does / why we need it:
Implements the Publish Dataset feature where both the dataset and it's parent collection are not published. In this case, the Publish Dataset Modal allows the user to publish both the dataset and collection at the same time.

**Note** that this feature depends on a js-dataverse PR that is not yet merged: https://github.com/IQSS/dataverse-client-javascript/pull/199.  When it is merged, package.json can be updated to use the latest 'alpha' version.

## Which issue(s) this PR closes:

- Closes #425 

## Special notes for your reviewer:
I'm using` feat/461-publish-collection` as the base branch because it contains the integration code to call the Publish Collection js-dataverse use case.

## Suggestions on how to test this:
**Step 1: Run the Development Environment**
1. Execute `npm i`.
2. Navigate with `cd packages/design-system && npm run build`.
3. Return with `cd ../../`.
4. Ensure you have a `.env` file similar to `.env.example`, with the variable `VITE_DATAVERSE_BACKEND_URL=http://localhost:8000`.
5. Navigate with `cd dev-env`.
6. Start the environment using `./run-env.sh unstable`.
7. To verify the environment, visit <http://localhost:8000> and check your local Dataverse installation.

**Step 2: Test the feature**
1. Create a collection. When it is created, it will have the 'Unreleased' status
2. Within the new collection, create a dataset. 
3. On the Dataset page,  Choose Publish Dataset,  and Publish.
4. On the Publish Dataset Modal, there will be a text asking if you want to publish the parent collection as well as the dataset.  It will contain a link to then parent collection
5. Choose 'Continue' to publish the dataset and collection
6. When the publish is completed, check that the dataset and the collection are both published.  The dataset should be Version 1.0, and the collection should no longer have the 'Unreleased' status

## Does this PR introduce a user interface change? If mockups are available, please link/include them here:

This is not a change from the JSF, but here is what the JSF modal looks like.  'Unpublished' is the name of the parent collection.
<img width="843" alt="Screenshot 2024-10-06 at 12 09 15 PM" src="https://github.com/user-attachments/assets/9df6f912-c3c6-4ff9-93d9-5536766292a6">

## Is there a release notes update needed for this change?:

## Additional documentation:
